### DISCUSSION TITLE
Can I Use Pagefind as a Library And Build Indexes Programatically From Custom Sources?

### DISCUSSION THREAD

**Discussion by zicklag** [Pagefind/pagefind#708] [Q&A] [answered]
Hey there! I'm building an app that has a use-case for scoped search of user data. I have the data stored in a specific format and when I update the data I want to build an index from it that can be used to search from the web frontend.

It doesn't look like there's an easy way to use pagefind to build custom indexes, but I wanted to ask to make sure.

Great job on this project, BTW!

**Comment by bglw** [comment, marked as answer]
Hi @zicklag — The [NodeJS API](https://pagefind.app/docs/node-api/) sounds like it could do what you need. Let me know if that's down the right track or not 🙂

**Reply by zicklag** [reply]
Oh, that's exactly what I'm looking for, thank you very much!
